queries/nim/highlights.scm

```diff
@@ -0,0 +1,756 @@
+
+; =============================================================================
+; @conditional         ; keywords related to conditionals (e.g. `if` / `else`)
+
+[
+  "if"
+  "when"
+  "case"
+  "elif"
+  "else"
+] @conditional
+
+(of_branch "of" @conditional)
+
+; =============================================================================
+; @repeat              ; keywords related to loops (e.g. `for` / `while`)
+
+[
+  "for"
+  "while"
+  "continue"
+  "break"
+] @repeat
+
+(for "in" @repeat)
+
+; =============================================================================
+; @label               ; GOTO and other labels (e.g. `label:` in C)
+
+(block
+  label: [
+    (identifier) @label
+    (accent_quoted (identifier) @label)
+  ])
+
+; =============================================================================
+; @include             ; keywords for including modules (e.g. `import` / `from` in Python)
+
+[
+  "import"
+  "include"
+  "export"
+] @include
+
+(import_from_statement "from" @include)
+
+(except_clause "except" @include)
+
+; =============================================================================
+; @exception           ; keywords related to exceptions (e.g. `throw` / `catch`)
+
+[
+  "try"
+  "except"
+  "finally"
+  "raise"
+] @exception
+
+; =============================================================================
+; @type            ; type or class definitions and annotations
+
+((identifier) @type
+  (#has-ancestor? @type type_expression)
+  (#not-has-ancestor? @type pragma_list))
+; NOTE: needs to be after @variable
+; NOTE: benchmarked with
+; `$ hyperfine -P version 1 3 "tree-sitter query -q $QUERIES/highlights{version}.scm $NIM_REPO/**/*.nim"`
+; with
+; 1. "no-priority" version,
+; 2. "priority and (_ (_ (_... nesting version" and
+; 3. "priority and has-ancestor version".
+; "no-priority" was the worst, the other two were almost the same,
+; but "has-ancestor" is superior, since it allows for infinite nesting.
+
+; generic types when declaring
+((generic_parameter_list
+  (parameter_declaration
+    (symbol_declaration_list
+      (symbol_declaration
+        name: [
+          (identifier) @type
+          (accent_quoted (identifier) @type)
+        ])))))
+
+; generic types when calling
+(call
+  function: (bracket_expression
+    right: (argument_list [
+      (identifier) @type
+      (_ (identifier) @type)
+      (_ (_ (identifier) @type))
+      (_ (_ (_ (identifier) @type)))
+      (_ (_ (_ (_ (identifier) @type))))
+      (_ (_ (_ (_ (_ (identifier) @type)))))
+      (_ (_ (_ (_ (_ (_ (identifier) @type))))))
+      (_ (_ (_ (_ (_ (_ (_ (identifier) @type)))))))
+      (_ (_ (_ (_ (_ (_ (_ (_ (identifier) @type))))))))
+      (_ (_ (_ (_ (_ (_ (_ (_ (_ (identifier) @type)))))))))
+      (_ (_ (_ (_ (_ (_ (_ (_ (_ (_ (identifier) @type))))))))))
+      (_ (_ (_ (_ (_ (_ (_ (_ (_ (_ (_ (identifier) @type)))))))))))
+      (_ (_ (_ (_ (_ (_ (_ (_ (_ (_ (_ (_ (identifier) @type))))))))))))
```

Comment:
I know this is to workaround tree-sitter/tree-sitter#880 but I don't think it's acceptable for nvim-treesitter

Reply:
Why not? What else am I supposed to do?

Reply:
Nothing for now, and wait until tree-sitter has figured this out.

Reply:
@lucario387 have a look at the latest commit, if you like that better

Reply:
Guess I have to pick a poison, priority or deep nesting 😅 

I prefer the former than the latter here.

@clason will decide.

Reply:
Correctness wise, this solution is even a little better since it catches everything no matter how deep.
Also it got rid of the the performance hungry `has-ancestor` and `not-has-ancestor` call on every `(identifier)`

Reply:
Priority is tricky to get right because of injections. You may have to tweak the exact values. 

But avoiding hacky nestings or expensive `ancestor` calls are more important.

Reply:
Tweak how? My injection queries look fine.

Reply:
You never know if someone doesn't inject Nim into a codeblock somewhere.

Reply:
At least make sure of this one injection: Nim codeblocks in Markdown files have correct highlighting

Reply:
was indeed not working but I think I fixed it